src/components/ActionMenu/components/Actions/Actions.tsx

```diff
@@ -35,32 +56,92 @@ export function Actions({actions = [], groups = []}: Props) {
     () => setActiveMenuGroup(undefined),
     [],
   );
-  const menuActions = [...actions, ...groups];
 
+  // eslint-disable-next-line react-hooks/exhaustive-deps
+  const handleResize = useCallback(
+    debounce(
+      () => {
+        if (!newDesignLanguage || !actionsLayoutRef.current) return;
+        setAvailableWidth(actionsLayoutRef.current.offsetWidth);
+      },
+      20,
+      {leading: false, trailing: true, maxWait: 40},
+    ),
+    [actionsLayoutRef],
+  );
+
+  useEffect(() => {
+    if (!actionsLayoutRef.current) return;
```

Comment:
This is for future consideration but it would be nice to see these split into little hooks that you include to make this a bit more readable and easier to follow what all the effects are for.